content/en/blog/_posts/2025-XX-XX-endpoints-deprecation.md

```diff
@@ -0,0 +1,230 @@
+---
+layout: blog
+title: "Continuing the transition from Endpoints to EndpointSlices"
+slug: endpoints-deprecation
+date: 2025-XX-XX
+draft: true
+author: >
+  Dan Winship (Red Hat)
+---
+
+Since the addition of [EndpointSlices] ([KEP-752]) as alpha in v1.15
+and later GA in v1.21, the
+Endpoints API in Kubernetes has been gathering dust. New Service
+features like [dual-stack networking] and [traffic distribution] are
+only supported via the EndpointSlice API, so all service proxies,
+Gateway API implementations, and similar controllers have had to be
+ported from using Endpoints to using EndpointSlices. At this point,
+the Endpoints API is really only there to avoid breaking end user
+workloads and scripts that still make use of it.
+
+As of Kubernetes 1.33, the Endpoints API is now officially deprecated,
+and the API server will return warnings to users who read or write
+Endpoints resources rather than using EndpointSlices.
+
+Eventually, the plan (as documented in [KEP-4794]) is to change the
+[Kubernetes Conformance] criteria to no longer require that clusters
+run the _Endpoints controller_ (which generates Endpoints objects
+based on Services and Pods) and the _EndpointSlice mirroring
```

Comment:
for the Endpointss controller the decision was clear and unanimous but not for the mirroring controller https://github.com/kubernetes/enhancements/pull/4975#discussion_r1922458035.

I think we should try to fail safe based on past experience with deprecations, is very unlikely the entire ecosystem changes and not having the mirroring controller will break clusters. 

Do you mind omitting the refrence to `the _EndpointSlice mirroring` on purpose, it is also an internal detail most people will not know it exists

Reply:
Sure (though ftr the discussion in the KEP was mostly about whether we could disable those controllers _by default_ rather than whether we could change conformance to give admins the _option_ to disable them).

Reply:
Conformance kind of means things enabled by default to guarantee portability... If y develop some project that create custom endpoints it will work in all conformance cluster because there has to be something that mirror the to slices in order to work ...

Reply:
I didn't understand that sorry, and I realized my mistake 
... I agree we moved the e2e and conformance tests to not depends on endpoints, and that is the right thing